### ISSUE TITLE
bug: can't find d2 executable

### ISSUE THREAD

**Issue by jcayzac** [terrastruct/d2-vscode#146]
The extension expects `d2` to live in the system path. It doesn't support `GOPATH`, and environment variables are not supported in the `D2.execPath` setting.

**Comment by suchwerk** [comment]
same here.

**Comment by BarryNolte** [comment]
Does it work if you set the d2 path in the settings to (as in the example above): "/Users/julien.cayzac/.go/bin/d2".  The logic is pretty simple, it takes that string from the settings and tries to execute it.

**Comment by suchwerk** [comment]
Yes this is working. I wounder why d2 is not found on the system path.

**Comment by Sayi** [comment]
I think we should use D2 WEB API calls by default instead of having to install d2 locally!
```
[17:17:01] - ************************************************************
[17:17:01] - D2 executable not found.
[17:17:01] - Make sure the D2 executable is installed and on your system PATH.
[17:17:01] - https://d2lang.com/tour/install
[17:17:01] - ************************************************************

```

**Comment by commandline-be** [comment]
My impression is this does not work because (sigh) the VSCode Extension search for D2 binary in the working directory, not in the actual path or system PATH.

**Comment by alixander** [comment]
i plan to release a d2 wasm/js package which will replace the need for an installed binary

**Comment by tannerxrh** [comment]
> i plan to release a d2 wasm/js package which will replace the need for an installed binary

any progress on this?
finding limitations with mermaid and want to have good diagrams in documentation without requiring others to install system level packages

**Comment by techzilla** [comment]
> i plan to release a d2 wasm/js package which will replace the need for an installed binary

I wanted to ask about this, getting approved binaries at our office is insanely difficult. 

You open to accepting a PR on this?
